src.py

```diff
@@ -0,0 +1,290 @@
+    ax[0].hlines(i, 0, len(matchimg_region_segmentation[i]))
+
+    if table_weight < counter:
+        table_weight = counter
+
+    counter = 0
+
+    for j in range(len(matchimg_region_segmentation[i])):
+        if matchimg_region_segmentation[i][j] == 2:
+            counter += 1
+        else:
+            if table_weight < counter:
+                table_weight = counter
+            counter = 0
+
+if table_weight == 0:
+    table_weight = counter
+
+print(table_weight)
+print(table_height)
+
+
+# Далее найдем на изображении стул. Для этого воспользуемся фильтром по цвету(со значениями, при которых стена не
+# попадает в значения маски), и локальным методом "mean". Далее применим эквивалент оператора и и исключем стену из
+# результата локального метода.
+
+
+fig, ax = plt.subplots(1, 3, figsize=(120, 60))
+
+BLUE_MIN = np.array([0, 40, 0],np.uint8)
+BLUE_MAX = np.array([180, 255, 255],np.uint8)
+
+hsv_img = cv2.cvtColor(img,cv2.COLOR_BGR2HSV)
+
+frame_threshed = cv2.inRange(hsv_img, BLUE_MIN, BLUE_MAX)
+
+gray_img = rgb2gray(img)
+local = (gray_img < threshold_local(gray_img, 501, method='mean'))
+
+res = []
+
+for i in range(len(frame_threshed)):
+    res.append([])
+    for j in range(len(frame_threshed[i])):
+        if (frame_threshed[i][j] == 255) and local[i][j]:
+            res[i].append(True)
+        else:
+            res[i].append(False)
+res = binary_closing(np.array(res), selem=np.ones((25, 25)))
+
+ax[0].imshow(frame_threshed)
+ax[1].imshow(local)
+ax[2].imshow(res)
+
+
+# Т.к. стул сложной формы, в общем случае невозможно точно определить высоту. Однако, обычно, высота спинки стула
+# примерно в 2 - 1.5 раза меньше высоты всего стула(в данном случае 1.8).
+
+
+chaire_height = 0
+
+if weidth > height:
+    for j in range(point_y, 0, -1):
+        if not res[j][point_x]:
+            break
+
+    top = j + 1
+    for j in range(top, len(res)):
+        if not res[j][point_x]:
+            break
+    bot = j - 1
+    chaire_height = (bot - top) * 1.8
+
+    print(chaire_height)
+else:
+    for i in range(point_x, 0, -1):
+        if not res[point_y][i]:
+            top = i
+            break
+    top = i + 1
+
+    for i in range(top, len(res[0])):
+        if not res[point_y][i]:
+            bot = i
+            break
+    bot = i - 1
+    chaire_weidth = (bot - top) * 1.8
+
+    print(chaire_weidth)
+
+
+# Для нахождения ширины стула воспользуемся тем, что самое широкое место сутла - душки по краям, почти сверху,
+# стула. Таким образом, для нахождения ширины стула лучше всего использовать то, он симметричен, метка находится
+# посередине стула, а значит, что нужно найти лишь душку с одной стороны.
+
+
+if weidth > height:
+    place_y = top + 100
+
+    for i in range(point_x, 0, -1):
+        if not res[place_y][i]:
+            break
+    x_tmp = i
+
+    for i in range(x_tmp, 0, -1):
+        if res[place_y][i]:
+            break
+
+    left = i
+    chaire_weidth = (point_x - left) * 2
```

Comment:
Что-то такое (это просто идейно, вероятно, стоит реализовать по-другому):
```suggestion
    chair_width = np.max(np.where(res, axis=0)[-1] - np.where(res, axis=0)[0])
```